lib/modules/collapseWhitespace.es6

```diff
@@ -19,8 +32,27 @@ export default function collapseWhitespace(tree, options, collapseType, tag) {
 
     tree.forEach((node, index) => {
         if (typeof node === 'string' && !isComment(node)) {
-            const isTopLevel = ! tag || tag === 'html' || tag === 'head';
-            node = collapseRedundantWhitespaces(node, collapseType, isTopLevel);
+            const prevNode = tree[index - 1];
+            const nextNode = tree[index + 1];
+            const prevNodeTag = prevNode && prevNode.tag;
+            const nextNodeTag = nextNode && nextNode.tag;
```

Comment:
`prevNodeTag` & `nextNodeTag` are required to determine if there is an `noTrimWhitespacesArroundElement` nearby the `tag`.